src/components/EventModal.tsx

```diff
@@ -0,0 +1,72 @@
+import { BOOKING_COLORS } from "_/constants"
+import { useAuth } from "_/contexts"
+import { Event } from "_/models"
+import { Button } from "./Button"
+
+interface Props{
+    event?: Event
+    openModal: boolean
+    closeModal: () => void
+}
+
+export function EventModal({event, openModal, closeModal} : Props){
+  const { isAdmin } = useAuth()
+    return (
+        openModal ? (
+            <>
+              <div
+                className="justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none"
+              >
+                <div className="relative w-auto my-6 mx-auto max-w-3xl">
+                  {/*content*/}
+                  <div className="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none">
+                    {/*header*/}
+                    <div className="flex items-start justify-between p-5 border-b border-solid border-slate-200 rounded-t">
+                      <h3 className="text-3xl font-semibold">
+                        {event?.modalTitle}
+                      </h3>
+                      <button
+                        className="p-1 ml-auto bg-transparent border-0 text-black opacity-5 float-right text-3xl leading-none font-semibold outline-none focus:outline-none"
+                        onClick={closeModal}
+                      >
+                        <span className="bg-transparent text-black opacity-5 h-6 w-6 text-2xl block outline-none focus:outline-none">
+                          ×
+                        </span>
+                      </button>
+                    </div>
+                    {/*body*/}
+                    <div className="relative p-6 flex-auto">
+                      <p className="my-4 text-slate-500 text-lg leading-relaxed">
+                        {event?.type === 'checkin' ?
+                          (<><b>Destino:</b> {event?.local}<br/></>)
+                          : (<><b>Status:</b> {event?.status}<br/></>)
+                        }
+                        {event?.modalDescription}
+                      </p>
+                    </div>
+                    {/*footer*/}
+                    <div className="flex items-center justify-end p-6 border-t border-solid border-slate-200 rounded-b">
+                      {event?.type === 'booking' && isAdmin && event.status === 'Pendente' && (
+                        // TODO: create function to actualy aprove and reject bookings
```

Comment:
I've just left this `TODO` tag here because I think these functions for updating the booking will be a little difficult, so I think it is better to leave it to another PR.

Reply:
I've added this logic here already. I originally did it on another branch, but this PR wasn't merged so I just added in this PR